### DISCUSSION TITLE
XLT: Server response could not be parsed

### DISCUSSION THREAD

**Discussion by srcd4** [Xceptance/XLT#315] [Q&A] [answered]
Hello!

I am working on some tests to do load testing. After submitting a form (registration), I get the following error `The server response could not be parsed as HTML.` and the test fails. Is there anything that I need to set in the properties file for this to work?

FYI The form is submitted correctly and I am able to log in as the user created by the script. 

Thank you!

**Comment by rschwietzke** [comment]
Thanks for your question but based on that description, we cannot diagnose it further. Use the [resultbrowser](https://docs.xceptance.com/xlt/load-testing/manual/440-result-browser/#using-the-result-browser) to see what is returned by that call when executing the scenario in your IDE. This might explain why the parsing does not work. We can go further from there. You can also share details with us (on a private channel) if you like.

**Please note**: This is not an official support channel. Responses times might vary.

**Reply by srcd4** [reply]
Hello! 

Thank you for the response. 

I just came back to this issue. I checked further and the response coming back from our server is `content-type: application/json` when the registration form is sent. When I examined the sample posters store for the same action (registration), it returns `content-type: text/html;charset=utf-8`

Is there anything I can do to get around this issue?

**Reply by wurzelkuchen** [reply, marked as answer]
Hi,

It sounds as if you try to submit the form by one of the `loadPageBy…` methods. This will always expect a full HTML document since it’s the equivalent of a page load in the browser.
When you get a JSON response the form submission is probably done via an XMLHttpRequest/Ajax request (XHR) by the JavaScript of your site. 
For such calls you’ll have to mimic this behavior by sending an HttpRequest with the same post and query parameters as well from your testsuite.
XLT brings the class `com.xceptance.xlt.engine.httprequest.HttpRequest` for such tasks. 

```
        WebResponse response = new HttpRequest()
            .baseUrl("https://mypage.com")
            .relativeUrl("path/to/form/submit")
            .method(HttpMethod.POST)
            .param("param", "value1")
            .param("param", "value2")
            .fire();

```

The parameters, method and URL have to be gathered from the page or (if reasonably possible) used from other sources like a CSV file etc.
If you started your project on the [posters-advanced-loadtest-suite](https://github.com/Xceptance/posters-advanced-loadtest-suite) you can also use the class `com.xceptance.loadtest.api.util.HttpRequest` which offers some more convenient methods, like a direct form submission:
```
WebResponse response = HttpRequest()
        .submitForm(form)
        .fire();

```
In this case you will only need to get the from from the page.

If you need the response content (e.g. to get a redirect URL), you can retrieve it from the response via the `response.getContentAsString();` method.

Hope this helps to get you on the right track. If you need further assistance let us know.

Best,
Bernd
